Autogenerate completion handler types

adetaylor opened the pull request:
#### 6dc6bb7a2e0f82169c3df3bfb8da63d98a030436
<pre>
Autogenerate completion handler types
<a href="https://bugs.webkit.org/show_bug.cgi?id=304244">https://bugs.webkit.org/show_bug.cgi?id=304244</a>
<a href="https://rdar.apple.com/166610257">rdar://166610257</a>

Reviewed by Richard Robinson.

This is a slight cleanup with no functional changes. Swift can&apos;t refer to
templated C++ types, so IPC message completion handlers require a &apos;using&apos;
statement. It turns out we can easily autogenerate them, which eliminates the
need for humans to write those &apos;using&apos; statements, and in the case of our IPC
message tester enables us to get rid of an entire header file.

* Source/WebKit/Modules/Internal/WebKitInternalCxx.h:
* Source/WebKit/Scripts/webkit/messages.py:
(message_to_completion_handler_using_declaration):
(generate_messages_header):
* Source/WebKit/Scripts/webkit/tests/TestWithSwiftMessages.h:
* Source/WebKit/Shared/IPCTesterReceiverSwiftTypes.h: Removed.
* Source/WebKit/Shared/WebKit-Swift.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/304578@main">https://commits.webkit.org/304578@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bca0d08ba05267ae6d2fed7c5fc53b5cdf57dfdc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135858 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8234 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47150 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/143572 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/87438 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6057cbb3-8b3a-4d31-bb6d-a0a5f5b0463a) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8892 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8080 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103825 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/71764 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e09e84d5-e090-4cb2-8c19-348914526ff9) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138804 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6432 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121770 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84702 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/746a2fd3-4540-4c7f-a23f-c81c3d5d1379) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/135206 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/6142 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3779 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4175 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115401 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/146320 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7919 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40525 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/112184 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7945 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6647 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112570 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28593 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6050 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118070 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/61825 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7966 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36151 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7699 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/71517 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7920 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7782 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/Shared/IPCTesterReceiverSwift.swift:
```diff
@@ -42,7 +42,7 @@ final class IPCTesterReceiverSwift {
     }
 
-    func asyncMessage(data: UInt32, completionHandler: AsyncMessageCompletionHandler) {
+    func asyncMessage(data: UInt32, completionHandler: CompletionHandlers.IPCTesterReceiverSwift.AsyncMessageCompletionHandler) {
```
we should have a way to make this just a normal `async` function at some point but that’s a not-now problem

adetaylor replied:
Agreed, this is rdar://166611537 (sorry there's no webkit tracker equivalent just yet)

adetaylor committed:
Autogenerate completion handler types
https://bugs.webkit.org/show_bug.cgi?id=304244
rdar://166610257

Reviewed by Richard Robinson.

This is a slight cleanup with no functional changes. Swift can't refer to
templated C++ types, so IPC message completion handlers require a 'using'
statement. It turns out we can easily autogenerate them, which eliminates the
need for humans to write those 'using' statements, and in the case of our IPC
message tester enables us to get rid of an entire header file.

* Source/WebKit/Modules/Internal/WebKitInternalCxx.h:
* Source/WebKit/Scripts/webkit/messages.py:
(message_to_completion_handler_using_declaration):
(generate_messages_header):
* Source/WebKit/Scripts/webkit/tests/TestWithSwiftMessages.h:
* Source/WebKit/Shared/IPCTesterReceiverSwiftTypes.h: Removed.
* Source/WebKit/Shared/WebKit-Swift.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/304578@main
